### PR TITLE
Code template for typescript APIs

### PR DESCRIPTION
**Pull request by tarang1998**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
 	"description": "An extension to help generate the minimal code to be used while implementing clean architecture.",
 	"version": "1.0.0",
 	"engines": {
-		"vscode": "^1.62.0"
+		"vscode": "^1.61.0"
```
Comment (tarang1998) [inline package.json]: @bssughosh 

I made this change

Comment (bssughosh) [reply package.json]: Why was this required?

File: package.json [hunk 2]
```diff
@@ -18,6 +18,10 @@
 			{
 				"command": "clean-arch-helper.newFlutterFeature",
 				"title": "CAH Flutter"
+			},
+			{
+				"command": "clean-arch-helper.newTSFeature",
```
Comment (bssughosh) [inline package.json:23]: @tarang1998 
U can add it to the Touch Bar part below too

File: src/code_generator_templates/ts/ts_gateway_layer_template.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
+
+
+
+export function getTSOrchestratorCodeTemplate(featureClassName: string, snakeCaseFeatureName: string): string {
+    const template = 
+    
+    `
+    import { container } from "../../../inversify.config";
```
Comment (bssughosh) [inline src/code_generator_templates/ts/ts_gateway_layer_template.ts:8]: This path might be wrong right?

File: src/code_generator_templates/ts/ts_gateway_layer_template.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    return template;
+};
+
+export function getTSDeserializerCodeTemplate(featureClassName: string): string {
+
+    const template = 
+    
+    `
+    import { injectable } from "inversify";
+    import { InvalidRequestArguments } from "../../../core/errors/errors";
```
Comment (bssughosh) [inline src/code_generator_templates/ts/ts_gateway_layer_template.ts:31]: Again this path might change

File: src/newTSFeature/codeGenerator.ts [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+
+	if (type == TSFeatureType.NEW_OPERATION) {
+
+        const targetPath = `${targetDirectory}/${snakeCaseFeatureName}`
+
+
+		generateDataLayerCode(`${targetPath}/data/repository`, featureClassName, snakeCaseFeatureName);
+
+		generateDomainLayerCode(`${targetPath}/domain/repository/`, featureClassName,snakeCaseFeatureName);
```
Comment (bssughosh) [inline src/newTSFeature/codeGenerator.ts:25]: gateway code is not being generated here

File: src/newTSFeature/codeGenerator.ts [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+	if (existsSync(filePath)) {
+		throw Error(`${fileName} already exists`);
+	}
+
+	writeFile(
+		filePath,
+		dataToWrite,
+		function () { },
+	);
+}
```
Comment (bssughosh) [inline src/newTSFeature/codeGenerator.ts]: This can be moved to utils folder right?

It is used in both places

File: package.json (hunk 1 above)
Comment (tarang1998) [reply package.json]: My VS code version was 1.61

So tried it with this 
It worked after this 

Ill update my VSCode version and update this

File: src/code_generator_templates/ts/ts_gateway_layer_template.ts (hunk 3 above)
Comment (tarang1998) [reply src/code_generator_templates/ts/ts_gateway_layer_template.ts:8]: Im assuming the folder would always be created inside the src file 
So it shoudn't be wrong

File: src/code_generator_templates/ts/ts_gateway_layer_template.ts (hunk 4 above)
Comment (tarang1998) [reply src/code_generator_templates/ts/ts_gateway_layer_template.ts:31]: Same here

File: src/newTSFeature/codeGenerator.ts (hunk 6 above)
Comment (tarang1998) [reply src/newTSFeature/codeGenerator.ts]: Cool
will do that

File: src/newTSFeature/codeGenerator.ts (hunk 5 above)
Comment (tarang1998) [reply src/newTSFeature/codeGenerator.ts:25]: Yup I thought it wont be needed 

all the APIs related to student operations are written in the main studentOperationsFolder

So there would be gateway code for each API (named them as EXISTING_OPERATIONS ) for now

Comment (bssughosh) [reply src/newTSFeature/codeGenerator.ts:25]: cool

**Pull request merged**